kvbc/test/categorization/kv_blockchain_test.cpp

```diff
@@ -545,57 +608,57 @@ TEST_F(categorized_kvbc, get_category) {
-  auto value = std::string("val");
+    VersionedUpdates ver_updates;
+    ver_updates.calculateRootHash(true);
+    ver_updates.addUpdate("ver_key1", "ver_val1");
+    ver_updates.addUpdate("ver_key2", VersionedUpdates::Value{"ver_val2", true});
+    updates.add("versioned", std::move(ver_updates));
+    ASSERT_EQ(block_chain.addBlock(std::move(updates)), (BlockId)1);
+  }
+  // Add block2
+  {
+    Updates updates;
+    BlockMerkleUpdates merkle_updates;
+    merkle_updates.addUpdate("merkle_key3", "merkle_value3");
+    merkle_updates.addDelete("merkle_key1");
+    updates.add("merkle", std::move(merkle_updates));
 
-  uint64_t state_root_version = 886;
-  BlockMerkleOutput mui;
-  mui.keys[key] = MerkleKeyFlag{false};
-  mui.root_hash = pHash;
-  mui.state_root_version = state_root_version;
+    VersionedUpdates ver_updates;
+    ver_updates.addUpdate("ver_key3", "ver_val3");
+    ver_updates.addDelete("ver_key2");
+    updates.add("versioned", std::move(ver_updates));
 
-  Block block{888};
-  block.add(cf, std::move(mui));
-  block.setParentHash(pHash);
+    VersionedUpdates ver_updates_2;
+    ver_updates_2.addUpdate("ver_key4", "ver_val4");
+    updates.add("versioned_2", std::move(ver_updates_2));
 
-  // Add it to the blockchain and load it
-  auto wb = db->getBatch();
-  tester.getBlockchain(block_chain).addBlock(block, wb);
-  db->write(std::move(wb));
-  auto last_block = tester.getBlockchain(block_chain).loadLastReachableBlockId();
-  ASSERT_EQ(*last_block, 888);
-  tester.getBlockchain(block_chain).setLastReachableBlockId(*last_block);
-
-  // write category kvs
-  db->createColumnFamily(cf);
-  auto db_key = v2MerkleTree::detail::DBKeyManipulator::genDataDbKey(std::string(key), state_root_version);
-  auto db_val = v2MerkleTree::detail::serialize(
-      v2MerkleTree::detail::DatabaseLeafValue{block.id(), sparse_merkle::LeafNode{std::string(value)}});
-  db->put(cf, db_key, db_val);
-  auto db_get_val = db->get(cf, db_key);
-  ASSERT_EQ(db_get_val.value(), db_val);
+    ImmutableUpdates immutable_updates;
+    immutable_updates.addUpdate("immutable_key2", {"immutable_val2", {"1", "2"}});
+    updates.add("immutable", std::move(immutable_updates));
+    ASSERT_EQ(block_chain.addBlock(std::move(updates)), (BlockId)2);
+  }
 
   // try to get updates
   ASSERT_THROW(block_chain.getBlockData(889), std::runtime_error);
   ASSERT_THROW(block_chain.getBlockData(887), std::runtime_error);
-  auto updates = block_chain.getBlockData(888);
-  auto variant = updates.kv[cf];
-  auto merkle_updates = std::get<BlockMerkleInput>(variant);
+  auto updates = block_chain.getBlockData(1);
+  auto variant = updates.kv["versioned"];
+  auto versioned_updates = std::get<VersionedInput>(variant);
   // check reconstruction of original kv
-  ASSERT_EQ(merkle_updates.kv[key], value);
+  ASSERT_TRUE(versioned_updates.calculate_root_hash);
+  auto expected = ValueWithFlags{"ver_val2", true};
+  ASSERT_EQ(versioned_updates.kv["ver_key2"], expected);
```

Comment:
Nit: I think using the generated `operator==()` for the VersionedInput  CMF type can simplify checks here.

Reply:
I think that until we add default initializers the generated operators are worthless